pkg/apis/servicecatalog/checksum/unversioned/checksum.go

```diff
@@ -38,6 +39,10 @@ func InstanceSpecChecksum(spec servicecatalog.InstanceSpec) string {
 	if spec.Parameters != nil {
 		specString += fmt.Sprintf("parameters:\n\n%v\n\n", string(spec.Parameters.Raw))
 	}
+	if spec.AlphaSecretParameters != nil {
+		// TODO: calc checksum of secret contents instead?
```

Comment:
I think technically you have to calculate both the checksum based on the spec and the checksum based on the values of the parameters.  The tricky part is that you have to calculate the checksum of the parameters based on the value of the parameters you set.  I think that implies a separate checksum field that the controller has to set that is based on the parameter values.

Note that if any of these values change due to user update, there's a bug in the controller because we do not implement update :) Someone should go and do that.